backend/gce/api.go

```diff
@@ -33,8 +33,14 @@ type gceAPI struct {
 	gceInstance    *compute.Instance
 }
 
+// limit auth scope to just the cloud-platform API's
+// https://developers.google.com/identity/protocols/oauth2/scopes
+func gceScopes() []string {
+	return []string{"https://www.googleapis.com/auth/cloud-platform"}
```

Comment:
I'm curious why you selected this particular scope? Looking under the link about the scopes, compute engine has a (supposedly) more narrow scope `https://www.googleapis.com/auth/compute` (https://developers.google.com/identity/protocols/oauth2/scopes#compute) - is it not enough for flannel?

Reply:
`https://www.googleapis.com/auth/compute` seems to work in my testing as well. I wonder if it should be configurable, just in case? 

FWIW, the `gcloud` command used here does suggest the `compute` scope: https://coreos.com/flannel/docs/latest/backends.html#gce

Reply:
It was really selected by customer S. I'm not sure compute engine is really the correct scope, as I would expect the route changes to lie somewhere else.

Reply:
> `https://www.googleapis.com/auth/compute` seems to work in my testing as well. I wonder if it should be configurable, just in case?
> 
> FWIW, the `gcloud` command used here does suggest the `compute` scope: https://coreos.com/flannel/docs/latest/backends.html#gce

Ok, I think going with compute might be better as it's presumably least privilege. My thought is I'd prefer not to make this configurable, as that creates more knobs that may need to be tuned, where presumably there is a best practice scope that needs the minimum required privileges.

Reply:
Yeah I guess the question is whether compute covers all the required APIs. I _think_ it does, but if not, we'd have to send another change through. In any case, using compute sounds like a good plan to me.

Reply:
As a reference the kubelet code (https://github.com/kubernetes/kubernetes/blob/release-1.17/staging/src/k8s.io/legacy-cloud-providers/gce/gce.go#L400) uses compute.NewService for it's client and looking at that source code (https://raw.githubusercontent.com/googleapis/google-api-go-client/master/compute/v1/compute-gen.go) the following scopes are added automatically

```
// NewService creates a new Service.
func NewService(ctx context.Context, opts ...option.ClientOption) (*Service, error) {
	scopesOption := option.WithScopes(
		"https://www.googleapis.com/auth/cloud-platform",
		"https://www.googleapis.com/auth/compute",
		"https://www.googleapis.com/auth/compute.readonly",
		"https://www.googleapis.com/auth/devstorage.full_control",
		"https://www.googleapis.com/auth/devstorage.read_only",
		"https://www.googleapis.com/auth/devstorage.read_write",
	)
	// NOTE: prepend, so we don't override user-specified scopes.
	opts = append([]option.ClientOption{scopesOption}, opts...)
	opts = append(opts, internaloption.WithDefaultEndpoint(basePath))
	client, endpoint, err := htransport.NewClient(ctx, opts...)
	if err != nil {
		return nil, err
	}
	s, err := New(client)
	if err != nil {
		return nil, err
	}
	if endpoint != "" {
		s.BasePath = endpoint
	}
	return s, nil
}
```

Reply:
Thanks, it's a great idea to compare against k8s. 

Based on that, it looks like the node will require cloud-platform anyways in the short term. However, when adopting the non-legacy cloud-controller-manager for GCE, presumably many of these scopes won't be needed. So as long as all the route injection we do via flannel is working with just compute, we're hopefully reduced the privilege requirements. 

Eventually we might want to move flannel towards a model where only the masters are injecting routes, similar to the way the cloud controllers are going, but we're not their yet.